packages/ra-ui-materialui/src/layout/Notification.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { useSelector, useDispatch } from 'react-redux';
+import Snackbar from '@material-ui/core/Snackbar';
+import Button from '@material-ui/core/Button';
+import { makeStyles, Theme } from '@material-ui/core/styles';
+import classnames from 'classnames';
+
+import {
+    hideNotification,
+    getNotification,
+    undo,
+    complete,
+    undoableEventEmitter,
+    useTranslate,
+} from 'ra-core';
+
+interface Props {
+    type?: string;
+    className?: string;
+    autoHideDuration?: number;
+}
+
+const useStyles = makeStyles((theme: Theme) => ({
+    confirm: {
+        backgroundColor: theme.palette.background.default,
+    },
+    warning: {
+        backgroundColor: theme.palette.error.light,
+    },
+    undo: {
+        color: theme.palette.primary.light,
+    },
+}));
+
+const Notification: React.FunctionComponent<Props> = ({
```

Comment:
I think you can add the SnackbarProps as we pass rest props to the underlying Snackbar component.

```suggestion
const Notification: React.FunctionComponent<Props & SnackbarProps> = ({
```

Reply:
Maybe not because the SnackbarProps has a non-optional [`open` property](url), which in our case is passed by Redux inside the component.

Reply:
`Omit<SnackbarPros, 'open'>` ?